staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -117,7 +117,7 @@ func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions n
 		Encoder:         encoder,
 		EmbeddedEncoder: embeddedEncoder,
 
-		Fixup: func(obj runtime.Object) runtime.Object {
+		FixupNonMutating: func(obj runtime.Object) runtime.Object {
 			result, err := transformObject(ctx, obj, options, mediaTypeOptions, scope, req)
```

Comment:
one question about whether asTable is actually non-mutating of the input

Reply:
what are the consequences if this fixup is mutating?

Reply:
It depends on the caller - if it assumes it won't mutate the object, then it violates its invariants.